docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,375 @@
+        ...
+```
+
+* **Use build modules to create aliases with arbitray names for targets**. Similar to the
+  previous example, some recipes use a build module with an alias to set an arbitray
+  target name. Let's see the example of the [tensorflow-lite
+  recipe](https://github.com/conan-io/conan-center-index/blob/03b24bf128cbf15d23ed988b8d8ca0c0ba87d307/recipes/tensorflow-lite/all/conanfile.py),
+  that uses this workaround to define a ``tensorflow::tensorflowlite`` target.
+  
+```python
+class TensorflowLiteConan(ConanFile):
+    name = "tensorflow-lite"
+    ...
+
+    def package_info(self):
+        # generate the target tensorflowlite::tensorflowlite
+        self.cpp_info.names["cmake_find_package"] = "tensorflowlite"
+        self.cpp_info.filenames["cmake_find_package"] = "tensorflowlite"
+        # this build module defines an alias tensorflow::tensorflowlite to the tensorflowlite::tensorflowlite generated target
+        self.cpp_info.build_modules["cmake_find_package"] = [os.path.join(self._module_subfolder, self._module_file)]
+        ...
+```
+
+To translate this information to the new model, just check which aliases are defined in the
+build modules and define those for the new model. In this case it should be enough with
+adding the ``tensorflow::tensorflowlite`` target with ``cmake_target_name`` to the root
+cpp_info (besides the ``cmake_file_name``property).
+
+```python
+class TensorflowLiteConan(ConanFile):
+    name = "tensorflow-lite"
+    ...
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "tensorflowlite"
+        self.cpp_info.filenames["cmake_find_package"] = "tensorflowlite"
+        self.cpp_info.build_modules["cmake_find_package"] = [os.path.join(self._module_subfolder, self._module_file)]
+
+        # set the tensorflowlite::tensorflowlite target name directly for CMakeDeps with no need for aliases
+        self.cpp_info.set_property("cmake_target_name", "tensorflow::tensorflowlite")
+        self.cpp_info.set_property("cmake_file_name", "tensorflowlite")
+        ...
+```
+
+### PkgConfigDeps
+
+The case of ``PkgConfigDeps`` is much more straight forward than the ``CMakeDeps`` case.
+This is because the current
+[pkg_config](https://docs.conan.io/en/latest/reference/generators/pkg_config.html)
+generator suports the new ``set_property`` model for most of the properties. Then, the current
+model can be translated to the new one without having to leave the old attributes in the
+recipes. Let's see an example:
+
+```python
+class AprConan(ConanFile):
+    name = "apr"
+    ...
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "apr-1"
+    ...
+```
+
+In this case, you can remove the ``.names`` attribute and just leave:
+
+```python
+class AprConan(ConanFile):
+    name = "apr"
+    ...
+    def package_info(self):
+        self.cpp_info.set_property("pkg_config_name",  "apr-1")
+    ...
+```
```

Comment:
Doesn't seem to be true anymore. `pkg_config` generator doesn't listen `set_property()` in conan 1.44.0